src/main/java/nl/novi/techiteasy1121/services/TelevisionService.java

```diff
@@ -1,72 +1,88 @@
+        this.ciModuleRepository = ciModuleRepository;
+        this.ciModuleService = ciModuleService;
     }
 
-    // Vanuit de repository kunnen we een lijst van Televisions krijgen, maar de communicatie container tussen Service en
-    // Controller is de Dto. We moeten de Televisions dus vertalen naar TelevisionDtos. Dit moet een voor een, omdat
-    // de translateToDto() methode geen lijst accepteert als argument, dus gebruiken we een for-loop.
     public List<TelevisionDto> getAllTelevisions() {
         List<Television> tvList = televisionRepository.findAll();
-        List<TelevisionDto> tvDtoList = new ArrayList<>();
-
-        for(Television tv : tvList) {
-            TelevisionDto dto = transferToDto(tv);
-            tvDtoList.add(dto);
-        }
-        return tvDtoList;
+        return transferTvListToDtoList(tvList);
     }
 
-    // Vanuit de repository kunnen we een lijst van Televisions met een bepaalde brand krijgen, maar de communicatie
-    // container tussen Service en Controller is de Dto. We moeten de Televisions dus vertalen naar TelevisionDtos. Dit
-    // moet een voor een, omdat de translateToDto() methode geen lijst accepteert als argument, dus gebruiken we een for-loop.
     public List<TelevisionDto> getAllTelevisionsByBrand(String brand) {
         List<Television> tvList = televisionRepository.findAllTelevisionsByBrandEqualsIgnoreCase(brand);
+        return transferTvListToDtoList(tvList);
+    }
+
+    public List<TelevisionDto> transferTvListToDtoList(List<Television> televisions){
         List<TelevisionDto> tvDtoList = new ArrayList<>();
 
-        for(Television tv : tvList) {
+        for(Television tv : televisions) {
             TelevisionDto dto = transferToDto(tv);
+            if(tv.getCiModule() != null){
+                dto.setCiModuleDto(ciModuleService.transferToDto(tv.getCiModule()));
+            }
+            if(tv.getRemoteController() != null){
+                dto.setRemoteControllerDto(remoteControllerService.transferToDto(tv.getRemoteController()));
+            }
             tvDtoList.add(dto);
         }
         return tvDtoList;
     }
 
-    // Deze methode is inhoudelijk hetzelfde als het was in de vorige opdracht. Wat verandert is, is dat we nu checken
-    // op optional.isPresent in plaats van optional.isEmpty en we returnen een TelevisionDto in plaats van een Television.
     public TelevisionDto getTelevisionById(Long id) {
-        Optional<Television> televisionOptional = televisionRepository.findById(id);
-        if (televisionOptional.isPresent()){
-            Television tv = televisionOptional.get();
+
+        if (televisionRepository.findById(id).isPresent()){
+            Television tv = televisionRepository.findById(id).get();
+            TelevisionDto dto =transferToDto(tv);
+            if(tv.getCiModule() != null){
+                dto.setCiModuleDto(ciModuleService.transferToDto(tv.getCiModule()));
+            }
```

Comment:
De toevoeging van de OneToOne relatie tussen Television en RemoteController is correct geïmplementeerd. Goede keuze om de Service-laag te gebruiken voor de logica in plaats van direct met de Repository te werken. Dit volgt de principes van de gescheiden verantwoordelijkheden.